refactor: feedback actions

chrismclarke opened the pull request:
PR Checklist

- [x] - Latest `master` branch merged
- [x] - PR title descriptive (can be used in release notes)

## Description
- Remove `enable` checkbox from feedback sidebar
- Add a _close_ button to the sidebar 
- Fix handling so that feedback mode is no longer enable whilst a feedback popup is open
- Add `feedback: open` and `feedback: close` actions to allow toggling feedback sidebar open and close
- Change previous `feedback: open: [name]` action (which opened a specific template) to `feedback: template: [name]` to avoid confusion
- Update [feature sheet](https://docs.google.com/spreadsheets/d/1IsjWr7NcPspsUWstdFKzTRD6iKmNxCutipQ5a0vvLsU/edit#gid=960980632) and documentation

## Git Issues

Closes #1790

## Screenshots/Videos

**Minor docs update**
![image](https://user-images.githubusercontent.com/10515065/219554503-c56fa32e-d8b9-45e3-9952-287536a90bb8.png)

**Updated feature sheet and demo**

[ParentApp (2).webm](https://user-images.githubusercontent.com/10515065/219554773-4809d470-8381-48bc-8197-9e910a515ef9.webm)

jfmcquade reviewed (approved):
The code looks good and the new functionality seems sensible

esmeetewinkel reviewed (approved):
Thanks, this resolves issue https://github.com/IDEMSInternational/parenting-app-ui/issues/1790. 

Would it be possible to spit out the current status of whether or not the feedback side menu is open in a protected boolean field? e.g. `_show_feedback_side_menu` = `TRUE` / ` FALSE`. That would enable me to spreadsheet-author a toggle for opening and closing the feedback side menu.

It is currently not possible to leave the `feature_feedback_debug` template with the feedback mode open (but there are work-arounds: manually editing the hyperlink or inserting a button that goes to the `home_screen` template). It might make sense to edit the functionality of the top header so that it navigates preserves whether or not the feedback side menu is open? 

[screen-capture (35).webm](https://user-images.githubusercontent.com/74557272/220130561-cd0e1271-6b6c-4a81-a421-a0b5eb7aae2a.webm)

chrismclarke commented:
@esmeetewinkel 

I've added a `_feedback_sidebar_open` field and updated the [feature sheet](https://docs.google.com/spreadsheets/d/1IsjWr7NcPspsUWstdFKzTRD6iKmNxCutipQ5a0vvLsU/edit#gid=960980632) to include toggling buttons, so should be good to go

[ParentApp (3).webm](https://user-images.githubusercontent.com/10515065/221061242-b9f3c6bd-0cd2-45c4-ac6a-50cd762d3e44.webm)
